src/esp/assets/ResourceManager.cpp

```diff
@@ -2849,10 +2902,54 @@ void ResourceManager::addComponent(
                  child,          // mesh transform node
                  visNodeCache,   // a vector of scene nodes, the visNodeCache
                  computeAbsoluteAABBs,  // compute absolute aabbs
-                 staticDrawableInfo);   // a vector of static drawable info
+                 staticDrawableInfo,    // a vector of static drawable info
+                 skinData);             // instance skinning data
   }
 }  // addComponent
 
+void ResourceManager::mapSkinnedModelToArticulatedObject(
+    const MeshTransformNode& meshTransformNode,
+    const RenderAssetInstanceCreationInfo& creationInfo,
```

Comment:
You only seem to be using the rig here, maybe just pass that pointer, instead of the entire creation info.